src/main/java/com/minecolonies/coremod/inventory/GuiHandler.java

```diff
@@ -1,30 +1,137 @@
 {
+    public static final int SCARECROW = 3;
+
     @Override
-    public Object getServerGuiElement(final int id, final EntityPlayer player, final World world, final int x, final int y, final int z)
+    public Object getServerGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z)
+    {
+        switch (ID)
+        {
+            case 1:
+            {
+                final ContainerItemHandler container = getServerElementForCapabilityProvider(player, world.getTileEntity(new BlockPos(x, y, z)));
+                if (container != null)
+                {
+                    return container;
+                }
+
+                break;
+                }
+            case 2:
+            {
+                final ContainerItemHandler container = getServerElementForCapabilityProvider(player, world.getEntityByID(x));
+                if (container != null)
+                {
+                    return container;
+                }
+
+                break;
+                }
+            case SCARECROW:
+            {
+                final BlockPos pos = new BlockPos(x, y, z);
+                final ScarecrowTileEntity tileEntity = (ScarecrowTileEntity) world.getTileEntity(pos);
+                return new Field(tileEntity, player.inventory, world, pos);
+            }
+            default:
+            {
+                return null;
+            }
+        }
+
+        return null;
+    }
+
+    @Nullable
+    private ContainerItemHandler getServerElementForCapabilityProvider(@NotNull final EntityPlayer player, @Nullable ICapabilityProvider provider)
     {
-        final BlockPos pos = new BlockPos(x, y, z);
-        final ScarecrowTileEntity tileEntity = (ScarecrowTileEntity) world.getTileEntity(pos);
-        return new Field(tileEntity, player.inventory, world, pos);
+        if (provider != null && provider.hasCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, null))
+        {
+            return new ContainerItemHandler(player.inventory, provider.getCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, null));
+        }
+        return null;
     }
 
     @Override
-    public Object getClientGuiElement(final int id, final EntityPlayer player, final World world, final int x, final int y, final int z)
+    public Object getClientGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z)
     {
-        final BlockPos pos = new BlockPos(x, y, z);
-        final ScarecrowTileEntity tileEntity = (ScarecrowTileEntity) world.getTileEntity(pos);
-        return new GuiField(player.inventory, tileEntity, world, pos);
+        switch (ID)
+        {
+            case 1:
+            {
+                final GuiItemHandler gui = getClientlementForCapabilityProvider(player, world.getTileEntity(new BlockPos(x, y, z)));
+                if (gui != null)
+                {
+                    return gui;
+                }
+
+                break;
+            }
+            case 2:
+            {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Extract this nested code block into a method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1199)